src/codec.rs

```diff
@@ -601,6 +601,90 @@ macro_rules! impl_for_non_zero {
 		)*
 	}
 }
+
+/// Encode the slice without prepending the len.
+///
+/// This is equivalent to encoding all the element one by one, but it is optimized for some types.
+pub(crate) fn encode_slice_no_len<T: Encode, W: Output>(slice: &[T], dest: &mut W) {
+	macro_rules! encode_to {
+		( u8, $slice:ident, $dest:ident ) => {{
+			let typed = unsafe { mem::transmute::<&[T], &[u8]>(&$slice[..]) };
+			$dest.write(&typed)
+		}};
+		( i8, $slice:ident, $dest:ident ) => {{
+			// `i8` has the same size as `u8`. We can just convert it here and write to the
+			// dest buffer directly.
+			let typed = unsafe { mem::transmute::<&[T], &[u8]>(&$slice[..]) };
+			$dest.write(&typed)
+		}};
+		( $ty:ty, $slice:ident, $dest:ident ) => {{
+			if cfg!(target_endian = "little") {
+				let typed = unsafe { mem::transmute::<&[T], &[$ty]>(&$slice[..]) };
+				$dest.write(<[$ty] as AsByteSlice<$ty>>::as_byte_slice(typed))
+			} else {
+				for item in $slice.iter() {
+					item.encode_to(dest);
+				}
+			}
+		}};
+	}
+
+	with_type_info! {
+		<T as Encode>::TYPE_INFO,
+		encode_to(slice, dest),
+		{
+			for item in slice.iter() {
+				item.encode_to(dest);
+			}
+		},
+	}
+}
+
+/// Encode the slice without prepending the len.
+///
+/// This is equivalent to encoding all the element one by one, but it is optimized in some
+/// situation.
+pub(crate) fn decode_vec_with_len<T: Decode, I: Input>(
```

Comment:
This code is just copy pasted from decode of `Vec<T>` so it can be used by bitvec.